bigtable/admin.go

```diff
@@ -808,6 +808,10 @@ type InstanceConf struct {
 	StorageType  StorageType
 	InstanceType InstanceType
 	Labels       map[string]string
+
+	// AutoscalingConfig configures the autoscaling properties on the cluster
+	// created with the instance. It is optional.
+	AutoscalingConfig AutoscalingConfig
```

Comment:
Whether this field should be a pointer or not is one of the key design decisions here. A zero value AutoscalingConfig isn't valid in the API, but whether we should catch that in the client or not is the question. Making it a pointer would alleviate the ambiguity while departing from the other value objects. @telpirion what do you think?

Reply:
And to answer your question -- this config will be for just the initial cluster.

Reply:
If I may, I think it would be best to keep this as a value field. For one, there's the parallelism with the other fields. For another, I think it's helpful to have the `isZero()` method to check whether the developer accidentally passed in a meaningless `AutoscalingConfig` (where all fields were set to zero).

Reply:
I think generally, in principle, it's best to let the service catch "bad" values rather than attempting to validate on the client side (been burned by this too many times!). However, I definitely see the rationale for keeping this field parallel to the existing fields. Either way is fine with me ultimately.

Reply:
I've been on both sides of this. I think a pointer aligns best with what a Go developer would expect. So I'll make this a pointer.